lib/nm/index.ts

```diff
@@ -0,0 +1,350 @@
+			return Bluebird.resolve();
+		} catch (err) {
+			throw formatError(500, `Could not forgetNetwork`, err);
+		}
+	}
+
+	listNearbyNetworks = async () => {
+		try {
+			const requestScanParams = [
+				['ssids', ['aay', [stringToArrayOfBytes('1')]]],
+			];
+			await this.requestScan(requestScanParams);
+			return this.getAccessPoints(this.devices.wifi.path);
+		} catch (err) {
+			if (_.isArray(err) && (err[0] === 'Scanning not allowed while already scanning' || err[0] === 'Scanning not allowed immediately following previous scan')) {
+				return this.getAccessPoints(this.devices.wifi.path);
+			}
+			throw formatError(500, `Could not listNearbyNetworks`, err);
+		}
+	}
+
+	getAccessPoints = async (wifiDevicePath) => {
+		const {AccessPoints} = await this.getObjectProperty(['org.freedesktop.NetworkManager.Device.Wireless', 'AccessPoints'], wifiDevicePath);
+		const getApsProperties = () => _.map(AccessPoints, this.getApProperties);
+		const rawAccessPoints = await Bluebird.all(getApsProperties());
+		const accessPoints = _.map(rawAccessPoints, (rawProps: any[]) => {
+			const props = _.reduce(rawProps, (acc, prop: any) => {
+				return {
+					...acc,
+					...prop,
+				};
+			}, {});
+			return props;
+		});
+		this.accessPoints = accessPoints;
+		return makeNetworksReadable(this.accessPoints);
+	}
+
+	getApProperties = async (path) => {
+		try {
+			const props = await Bluebird.all([
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Ssid'], path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Flags'], path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'WpaFlags'], path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'RsnFlags'], path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Frequency'], path),
+			]);
+			return props;
+		} catch (err) {
+			throw formatError(500, `Could not get ap ${path} property`, err);
+		}
+	}
+
+	getCurrentNetwork = async () => {
+		try {
+			const [key, [connections]] = await this.getActiveConnections();
+			const getConnectionsType = () => _.map(connections, _.partial(this.getObjectProperty, ['org.freedesktop.NetworkManager.Connection.Active', 'Type']));
+			const results = await Bluebird.all(getConnectionsType());
+			const wifiConnection: any = _.filter(results, ({Type}) => Type === '802-11-wireless')[0];
+			if (_.isUndefined(wifiConnection)) {
+				return;
+			}
+			const {Id} = await this.getObjectProperty(['org.freedesktop.NetworkManager.Connection.Active', 'Id'], wifiConnection.path);
+			return Id;
+		} catch (err) {
+			if (err === 404) {
+				throw formatError(404, 'You\'re not currently connected to a wireless network');
+			}
+			throw formatError(500, `Could not getCurrentNetwork`, err);
+		}
+	}
+}
+
+function formatError(type: number = 400, message: string, err: any = {}) {
+	return {type, message, err};
```

Comment:
@jhermsmeier  Yeah I know, I tried adding that but it says that nor `code` or `data` exist on type `Error` so I figured it'd be better to just keep it as it is

Reply:
Grrr, TypeScript... but we *really* don't want to throw anything that's not an instance of `Error` – that leads to very wild side-effects and breakage.
@pimterry @LucianBuzzo any recommendations? Do you just subclass Error, or how would you preferably do this in TypeScript?

Reply:
@thundron best thing I can come up with would be to follow https://github.com/Microsoft/TypeScript-wiki/blob/master/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work and subclass Error as per their recommendation

Reply:
We mostly use https://github.com/resin-io-modules/typed-error everywhere else, which gives you nicely typed error subclasses very easily. You can add any extra fields or functionality you need as part of your subclass there very easily.

Reply:
Looks nice, but I keep getting an error saying that I cannot invoke `TypedError` without using `new`, which I am doing. I think it might be a compilation error since I see this is something that shows up when trying to extend ES6 classes in ES5, or something like this.
To make it work for now, I extended the basic `Error` class.